docs/src/controller.md

```diff
@@ -0,0 +1,120 @@
+# Custom Pod Controller
+
+Kubernetes uses the
+[Controller pattern](https://kubernetes.io/docs/concepts/architecture/controller/)
+to align the current cluster state with the desired one.
+
+Stateful applications are usually managed with the
+[`StatefulSet`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/)
+controller, which allows to create and align a set of Pods created with the
+same specification and maintaining a sticky identity.
+
+CloudNativePG, instead of relying on the `StatefulSet` controller to manage
+the PostgreSQL instances, implements its own custom controller. This design
+choice, while bringing more complexity in the implementation, provided the
+operator with more flexibility on how we manage the cluster while being
+transparent on the topology of PostgreSQL clusters.
+
+PostgreSQL's operators can be certainly designed around the `StatefulSet`
+behavior (and many are indeed) but CloudNativePG avoid this tight
+coupling. This may surprise new users.
+
+Like many choices in the design realm, different ones lead to different
+compromises. The next sections discuss a few of the points where we believe
+this design choice has made the implementation of CloudNativePG
+more reliable and easy to understand.
+
+## Primary Instances versus Replicas
+
+The `StatefulSet` controller is designed to create a set of Pods
+from just one template. Given that we use one `Pod` per PostgreSQL instance,
+we have two kinds of Pods:
+
+- one for the primary instance, and
+- the other ones for replicas.
+
+While the operator is using the same spec for every Pod and the role
+is managed using labels, the rolling deployment mechanism depends on the
+operation that triggered it.
+
```

Comment:
I think it could be a bit confusing to introduce the `role` concept suddenly.

We could just say:

```suggestion
The primary and replica pods enact different rolling deployment mechanisms depending on the executed operation
```